hws/src/dr_ste.py

```diff
@@ -303,11 +315,12 @@ def dump_raw_ste(self, verbosity, tabs):
 
         return _str
 
-    def tree_print(self, verbosity, tabs, prefix=None):
+    def tree_print(self, verbosity, tabs, prefix, expected_miss_addr):
         _str = tabs + self.dump_str(verbosity, prefix)
         tabs = tabs + TAB
 
         _str += self.dump_fields(verbosity, tabs)
+        _str += self.dump_miss(verbosity, tabs, expected_miss_addr)
```

Comment:
what is the unexpected miss address, all the miss addresses are outputted in the matcher dump line

Reply:
We _assume_ that the STE memory matches what is expected based on the STC/RTC content. However, this tool is a debug tool; we use it to investigate problems where things are not working as expected. It's important that it tells us about such unexpected things rather than hiding the problem. If everything is as expected, then this code modification does not print anything new. However, if things are not as expected, then the code modification enhances the tool to print out what is actually present in the STE, to let the user know that something strange is going on. This can help them debug their issue. Note that this is explained in the individual commit description.